python/common/business-registry-account/src/business_account/AccountService.py

```diff
@@ -0,0 +1,204 @@
+from flask import current_app
+from http import HTTPStatus
+import json
+import requests
+from typing import Dict
+
+class AccountService:
+    """Wrapper to call Authentication Services.
+
+    @TODO Cache and refresh / retry token as needed to reduce calls.
+    """
+
+    BEARER: str = 'Bearer '
+    CONTENT_TYPE_JSON = {'Content-Type': 'application/json'}
+
+    try:
+        timeout = int(current_app.config.get('ACCOUNT_SVC_TIMEOUT', 20))
+    except Exception:
+        timeout = 20
+
+    @classmethod
+    def get_bearer_token(cls):
+        """Get a valid Bearer token for the service to use."""
+        token_url = current_app.config.get('ACCOUNT_SVC_AUTH_URL')
+        client_id = current_app.config.get('ACCOUNT_SVC_CLIENT_ID')
+        client_secret = current_app.config.get('ACCOUNT_SVC_CLIENT_SECRET')
+
+        data = 'grant_type=client_credentials'
+
+        # get service account token
+        res = requests.post(url=token_url,
+                            data=data,
+                            headers={'content-type': 'application/x-www-form-urlencoded'},
+                            auth=(client_id, client_secret),
+                            timeout=cls.timeout)
+
+        try:
+            return res.json().get('access_token')
+        except Exception:
+            return None
+
+    @classmethod
+    # pylint: disable=too-many-arguments, too-many-locals disable=invalid-name;
+    def create_affiliation(cls, account: int,
```

Comment:
Are the 'affiliation' functions used outside the legal-api?

Reply:
Data flow, filer and a lear support notebook (business-ops) at least use the affiliation

Reply:
Can you update this with latest code. You are missing flags parameter in this function https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/services/bootstrap.py#L151

Reply:
Updated, note that this was up to date when I started the work